### ISSUE TITLE
NoneType' object is not subscriptable

### ISSUE THREAD

**Issue by Grzy7316** [Jelomite/horrible-downloader#29]
Command I Ran: 
horrible-downloader -d "Kateikyoushi Hitman Reborn!" -e 8 -o ~/mnt/s/HorribleDownloader

Error I Received 
Traceback (most recent call last):
  File "/home/steve/.local/bin/horrible-downloader", line 11, in <module>
    sys.exit(cli())
  File "/home/steve/.local/lib/python3.6/site-packages/HorribleDownloader/cmd.py", line 369, in cli
    main(args)
  File "/home/steve/.local/lib/python3.6/site-packages/HorribleDownloader/cmd.py", line 225, in main
    title = Parser().get_episodes(args.download)[0]["title"]
  File "/home/steve/.local/lib/python3.6/site-packages/HorribleDownloader/parser.py", line 88, in get_episodes
    showid = self._get_show_id(show)
  File "/home/steve/.local/lib/python3.6/site-packages/HorribleDownloader/parser.py", line 19, in _get_show_id
    key = fuzzy_match.extractOne(show, self.shows.keys())[0]
TypeError: 'NoneType' object is not subscriptable

**Comment by mtshrmn** [comment]
Thank you for submitting an issue, I'll try to solve this as soon as possible

**Comment by mtshrmn** [comment]
I'm unable to recreate the problem. When I run this command I get the following:

```sh
$ horrible-downloader -d "Kateikyoushi Hitman Reborn!" -e 8 -o ~/mnt/s/HorribleDownloader
Traceback (most recent call last):
  File "/usr/bin/horrible-downloader", line 8, in <module>
    sys.exit(cli())
  File "/home/suerflowz/.local/lib/python3.8/site-packages/HorribleDownloader/cmd.py", line 369, in cli
    main(args)
  File "/home/suerflowz/.local/lib/python3.8/site-packages/HorribleDownloader/cmd.py", line 225, in main
    title = Parser().get_episodes(args.download)[0]["title"]
IndexError: list index out of range
```

#### There **is** a bug here, but It's not the same one that you're describing.
Can you recreate this bug again? If so, please provide more info about your python version and OS so I could recreate this on my machine.

**Comment by ProgrammingPleb** [comment]
Hi, while I'm not the one who opened the issue, I am also having the same issue, here is the output:
```
PS C:\Users\username\Documents\Projects\horriblesubs_srt> horrible-downloader -d "Fruits Basket S2 (2019)" -e 3 -o ~/Videos/Anime
Traceback (most recent call last):
  File "c:\users\username\appdata\local\programs\python\python38-32\lib\runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\username\appdata\local\programs\python\python38-32\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\username\AppData\Local\Programs\Python\Python38-32\Scripts\horrible-downloader.exe\__main__.py", line 7, in <module>
  File "c:\users\username\appdata\local\programs\python\python38-32\lib\site-packages\HorribleDownloader\cmd.py", line 369, in cli
    main(args)
  File "c:\users\username\appdata\local\programs\python\python38-32\lib\site-packages\HorribleDownloader\cmd.py", line 225, in main
    title = Parser().get_episodes(args.download)[0]["title"]
  File "c:\users\username\appdata\local\programs\python\python38-32\lib\site-packages\HorribleDownloader\parser.py", line 88, in get_episodes
    showid = self._get_show_id(show)
  File "c:\users\username\appdata\local\programs\python\python38-32\lib\site-packages\HorribleDownloader\parser.py", line 19, in _get_show_id
    key = fuzzy_match.extractOne(show, self.shows.keys())[0]
TypeError: 'NoneType' object is not subscriptable
```

As per your request, I'm getting this error on Python 3.8.1 and my current machine is running on Windows 10. The command itself was run on Powershell Core (Powershell 6.2.3).
Hope this helps!

P.S. based on your machine commands, I might test this script on Ubuntu 20.04 and see if something else comes up.

**Comment by mtshrmn** [comment]
Okay, I found the problem.
The machine I was trying to recreate the problem on had an outdated version of `horrible-downloader` (version 1.0.5).

Once I've updated it to the current version, I can easily recreate the issue.
This narrows down the problem to version 1.0.6 only.

Meanwhile you can downgrade the version to 1.0.5 and upgrade to the latest update as soon as I publish a new version.

**Comment by ProgrammingPleb** [comment]
Thank you for the quick reply!
I will try this out as soon as my machine completes this video rendering I’m doing currently.
I’ve looked through the commits between the two versions and I think the pull request #28 must have something to do with this issue?

**Comment by mtshrmn** [comment]
You're exactly right, the problem is that `rapidfuzz` doesn't behave **exactly** like `fuzzy_wuzzy`. 
I know how to fix it, but I would prefer that rapidfuzz would fix their problem as well. 

That said, I need to write better tests so issues like this one would've been caught earlier.

**Comment by ProgrammingPleb** [comment]
I hope that the latest release would be pulled for the time being to avoid other users to have the same issue as well.
Other than that, thank you for your hard work in maintaining this library!
This library is great, having more customization and even includes downloading, which is something I couldn’t find in other libraries.
Keep up the good work!

**Comment by mtshrmn** [comment]
#### The current version (1.0.6) is now yanked. 
for more details about yanking a package - https://pypi.org/help/#yanked

**Comment by mtshrmn** [comment]
Because the previous version (1.0.5) is stable, This patch won't be published on pypi. In one month this will be published as version 1.1.0 with more features and bugfixes.
If anyone wishes to use this patch, please install it manually from this [commit](https://github.com/Jelomite/horrible-downloader/commit/71ca27c8d227cbc172578947a3b3fab778b47544)
